sqlmesh/core/snapshot/definition.py

```diff
@@ -1318,6 +1320,8 @@ def create(
         deployability_mapping: t.Dict[SnapshotId, bool] = {}
         representative_shared_version_ids: t.Set[SnapshotId] = set()
 
+        start_date_cache: t.Optional[t.Dict[str, datetime]] = {}
```

Comment:
this uses a new cache for ever snapshot, can we reuse this across all snapshots? i'm afraid this could be expensive

Reply:
The cache is shared between all snapshots that are part of the index